### ISSUE TITLE
export to PDF

### ISSUE THREAD

**Issue by ynohat** [ynohat/mdtk#8]
currently recommended approach

reveal packager: use decktape
typora packager: use chrome+print to pdf

would be nice to have an integrated approach

**Comment by lukaszczerpak** [comment]
`typora` packager is used for labs typically and html format it generates is the best to share with others, as pdf can mess up especially there are fragments to copy/paste to terminal (non-printable characters, line breaks, etc..). 
On a separate note I've heard good opinions about https://wkhtmltopdf.org (another option for typora I believe).

**Comment by ynohat** [comment]
Wkhtmltopdf just converts from html to pdf (used it extensively in a previous life). Typora is not actually used by the typora packager, it’s just a reference to the source of the stylesheet mdtk stole. We should certainly rename it before 1.0, but that’s a different issue I think.

I believe an option to export to pdf in one command has value in some scenarios where convenient sharing or archiving is more important than the ability to copy paste safely.

Thoughts on this?

Anthony Hogg

**Comment by lukaszczerpak** [comment]
I have good results with puppeteer for typora packager. Zero issues, great results.

**Comment by ynohat** [comment]
@lukaszczerpak thanks a lot for that.

Currently I'm getting reports that sometimes mdtk is not easy to install (lots of errors related to sharp and canvas that have specific install prereqs, can't remember the details.

Puppeteer adds chromium to the dependencies, and will only serve the booklet/typora packager (at least as a quick thing, I think we could implement a puppeteer based renderer for revealjs as well, but that will be a much larger effort).

Can you share your approach for rendering booklets using puppeteer outside of mdtk here?

**Comment by lukaszczerpak** [comment]
I have `puppeteer-cli` package installed (which requires `puppeteer` as a dependency), then I simply run the following:

```
puppeteer print lab1.html lab1.pdf
```

**Comment by ynohat** [comment]
Right, so `npx puppeteer print lab1.html lab1.pdf` should work even without adding it to the dependencies of the package.

Let me think about this a little bit.
